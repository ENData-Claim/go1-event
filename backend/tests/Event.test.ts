import { createTestContext } from './__helpers'

const ctx = createTestContext()

describe('Event', () => {
  test('eventConnection query', async () => {
    const result = await ctx.client.send(`
      query EventConnectionQuery {
        eventConnection(first: 10) {
          pageInfo {
            endCursor
            hasNextPage
          }
          edges {
            cursor
            node {
              id
              title
              time
              image
            }
          }
        }
      }
    `)
    result.eventConnection.edges.forEach(({ node }: any) => {
      expect(node).toMatchSnapshot({
        id: expect.any(String)
      })
    })

    const nodeId = result.eventConnection.edges[0].node.id
    const nodeResult = await ctx.client.send(
      `
        query EventNodeQuery($id: ID!) {
          node(id: $id) {
            id
            __typename
            ... on Event {
              title
              time
              image
            }
          }
        }
      `,
      { id: nodeId }
    )
    expect(nodeResult.node).toHaveProperty('id')
    expect(nodeResult.node).toHaveProperty('title')
    expect(nodeResult.node).toHaveProperty('time')
    expect(nodeResult.node).toHaveProperty('image')
  })
})