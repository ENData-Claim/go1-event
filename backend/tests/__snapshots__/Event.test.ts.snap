// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event eventConnection query 1`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-22T02:30:00.000Z",
  "title": "Place 1",
}
`;

exports[`Event eventConnection query 2`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 2",
}
`;

exports[`Event eventConnection query 3`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 3",
}
`;

exports[`Event eventConnection query 4`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 4",
}
`;

exports[`Event eventConnection query 5`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 5",
}
`;

exports[`Event eventConnection query 6`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 6",
}
`;

exports[`Event eventConnection query 7`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 7",
}
`;

exports[`Event eventConnection query 8`] = `
Object {
  "id": Any<String>,
  "image": "http://example.com/image.png",
  "time": "2018-07-24T02:30:00.000Z",
  "title": "Place 8",
}
`;
